packages/ui/aurora/src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { forwardRef, PropsWithChildren } from 'react';
+
+import { useThemeContext } from '../../hooks';
+import { ThemedClassName } from '../../util';
+import { Button } from '../Buttons';
+import { Input } from '../Input';
+
+const COMBOBOX_NAME = 'ComboBox';
+
+type ScopedProps<P> = P & { __scopeComboBox?: Scope };
+
+type ValueAdapter<T> = (value: T) => { id: string; text: string };
+
+type ComboBoxContextValue<T> = {
+  items: T[];
+  adapter: ValueAdapter<T>;
+} & UseComboboxReturnValue<T>;
+
+const usePopperScope = createPopperScope();
+const [createComboBoxContext] = createContextScope(COMBOBOX_NAME, [createPopperScope]);
+const [ComboBoxProvider, useComboBoxContext] = createComboBoxContext<ComboBoxContextValue<any>>(COMBOBOX_NAME);
+
+type ComboBoxRootProps<T> = ThemedClassName<
+  ScopedProps<
+    PropsWithChildren<{
+      placeholder?: string;
+      items?: T[];
+      value?: T;
+      adapter: ValueAdapter<T>;
+      onChange?: (value: T | undefined) => void;
+      onInputChange?: (text?: string) => void;
+    }>
+  >
+>;
+
+/**
+ * Type-ahead combobox.
+ * https://www.downshift-js.com
+ */
+const ComboBoxRoot = <T,>({
+  __scopeComboBox,
+  children,
+  classNames,
+  placeholder,
+  items,
+  value,
+  adapter,
+  onChange,
+  onInputChange,
+}: ComboBoxRootProps<T>) => {
+  const { tx } = useThemeContext();
+  const popperScope = usePopperScope(__scopeComboBox);
+
+  // https://www.downshift-js.com/use-combobox
+  const comboProps = useCombobox<T>({
+    items: items ?? [],
+    selectedItem: value ?? null,
+    itemToString: (selectedItem) => (selectedItem ? adapter(selectedItem).text : ''),
+    onInputValueChange: ({ inputValue }) => onInputChange?.(inputValue),
+    onSelectedItemChange: ({ selectedItem }) => onChange?.(selectedItem === null ? undefined : selectedItem),
+  });
+
+  return (
+    <PopperPrimitive.Root {...popperScope}>
+      <ComboBoxProvider scope={__scopeComboBox} adapter={adapter} items={items ?? []} {...comboProps}>
+        <div className={tx('combobox.root', 'combobox__root', {}, classNames)}>
+          <ComboBoxAnchor placeholder={placeholder} />
+          <ComboBoxContent />
+        </div>
+      </ComboBoxProvider>
+    </PopperPrimitive.Root>
+  );
+};
+
+//
+// Anchor
+//
+
+const ANCHOR_NAME = 'ComboBoxAnchor';
+
+type ComboBoxAnchorProps = ThemedClassName<
+  ScopedProps<
+    PropsWithChildren<{
+      placeholder?: string;
+    }>
+  >
+>;
+
+const ComboBoxAnchor = forwardRef<HTMLDivElement, ComboBoxAnchorProps>(
+  ({ __scopeComboBox, classNames, placeholder }: ComboBoxAnchorProps, forwardedRef) => {
+    const { tx } = useThemeContext();
+    const popperScope = usePopperScope(__scopeComboBox);
+    const { getInputProps, getToggleButtonProps, isOpen } = useComboBoxContext(ANCHOR_NAME, __scopeComboBox);
+
+    return (
+      <PopperPrimitive.Anchor asChild {...popperScope} ref={forwardedRef}>
+        <div className='flex items-center gap-1'>
```

Comment:
`flex` should almost never be default, and classnames mustn’t be baked-in like this, this needs to be refactored to use `tx`.